tests/test_upload.py

```diff
@@ -100,3 +104,28 @@ def test_get_config_old_format(tmpdir):
             "more info: "
             "https://docs.python.org/distutils/packageindex.html#pypirc\n"
         ).format(pypirc)
+
+
+def test_skip_existing_skips_files_already_on_PyPI(monkeypatch):
+    response = mock.Mock(spec=upload.requests.models.Response)
+    session = mock.Mock(spec=upload.requests.sessions.Session)
+
+    monkeypatch.setattr(upload.requests, 'session', lambda: session)
+    monkeypatch.setattr(upload.os.path, 'exists', lambda args: True)
```

Comment:
Good point. I had started down the road of mocking this but don't need to any more since I'm using a fixture.